src/kudu/consensus/metadata.proto

```diff
@@ -104,6 +104,10 @@ message RaftPeerPB {
   optional MemberType member_type = 2;
   optional HostPortPB last_known_addr = 3;
 
+  // Optional node to receive replicate messges from, instead of directly from
+  // the leader.
+  optional bytes proxy_from = 6;
```

Comment:
Are we allowing RaftPeerPB to diverge on all nodes of the ring?
is automation/going to hit the nodes independently to set this

Reply:
The proxy node definitions will be eventually consistent over the course of a config change operation.

Reply:
An easy way to not have to go through consensus is to pull this out from RaftPeerPB and make a separate
protobu called PeerUUID->ProxyFrom map. And assume that its someone else's concern to thread that
map to you. should be relatively simple change.